app/frontend/users/UserEdit.res

```diff
@@ -609,10 +631,56 @@ let make = (
                       id="new_password"
                       value=state.newPassword
                       onChange={event =>
-                        send(UpdateNewPassword(ReactEvent.Form.target(event)["value"]))}
+                        send(UpdateNewPassword(ReactEvent.Form.target(event)["value"], schoolName))}
                       className="appearance-none block text-sm w-full shadow-sm border border-gray-300 rounded px-4 py-2 my-2 leading-relaxed focus:outline-none focus:border-transparent focus:ring-2 focus:ring-focusColor-500"
                       placeholder={t("new_password_placeholder")}
                     />
+                    {switch state.newPasswordAnalysis {
+                    | None => <div className="h-6" />
+                    | Some(zxcvbn) =>
+                      <div className="h-6">
+                        <div className="flex justify-between items-center">
+                          <p className="text-xs text-gray-400 font-inter">
+                            {t("password_strength")->str}
+                          </p>
+                          <div className="flex items-center gap-1 mt-1">
+                            <span className="text-xs text-gray-400 pe-2 text-right rtl:text-left">
+                              {zxcvbn->Zxcvbn.label->str}
+                            </span>
+                            <span
+                              className={`rounded-md h-1 bg-${Zxcvbn.score(zxcvbn) >= 0
+                                  ? Zxcvbn.color(zxcvbn)
+                                  : "gray"}-500 w-10`}
+                            />
+                            <span
+                              className={`rounded-md h-1 bg-${Zxcvbn.score(zxcvbn) > 1
+                                  ? Zxcvbn.color(zxcvbn)
+                                  : "gray"}-500 w-10`}
+                            />
+                            <span
+                              className={`rounded-md h-1 bg-${Zxcvbn.score(zxcvbn) > 2
+                                  ? Zxcvbn.color(zxcvbn)
+                                  : "gray"}-500 w-10`}
+                            />
+                            <span
+                              className={`rounded-md h-1 bg-${Zxcvbn.score(zxcvbn) > 3
+                                  ? Zxcvbn.color(zxcvbn)
+                                  : "gray"}-500 w-10`}
+                            />
```

Comment:
Forming class names in this manner is a no-no. This makes it impossible for the CSS purge step to detect. Write full class names, always.

i.e., instead of:

```res
`bg-${Zxcvbn.score(zxcvbn) > 2 ? Zxcvbn.color(zxcvbn) : "gray"}-500`
```

write something like:

```res
Zxcvbn.colorClass(zxcvbn) // which returns the full class name.
```